staging/src/k8s.io/apiserver/pkg/util/flowcontrol/fairqueuing/queueset/queueset.go

```diff
@@ -337,6 +337,16 @@ func (req *request) Finish(execFn func()) bool {
 		}()
 
 		execFn()
+
+		// Sleep for additional latency to block concurrency units.
+		if req.additionalLatency > 0 {
+			// TODO: In the current form this means that it's actually contributing to the
```

Comment:
@wojtek-t  yes, the additional latency should not have any impact on the user experience, i have a work in progress where we can achieve this without having any impact on user experience - https://github.com/kubernetes/kubernetes/pull/103240

The second commit https://github.com/kubernetes/kubernetes/pull/103240/commits/2a23c504622ffed2011c4cbf4997ca249338c665 sleeps outside the goroutine that executes the request handler.

The first commit make room for additional latency into width - i can remove it if we don't need it

Reply:
Great - I commented on that PR, it looks cool.